pkg/apis/kops/cluster.go

```diff
@@ -396,6 +397,10 @@ type EtcdClusterSpec struct {
 	Backups *EtcdBackupSpec `json:"backups,omitempty"`
 	// Manager describes the manager configuration
 	Manager *EtcdManagerSpec `json:"manager,omitempty"`
+	// MemoryRequest specifies the memory requests of each etcd container in the cluster.
+	MemoryRequest resource.Quantity `json:"memoryRequest,omitempty"`
+	// CPURequest specifies the cpu requests of each etcd container in the cluster.
+	CPURequest resource.Quantity `json:"cpuRequest,omitempty"`
```

Comment:
Can you set these to `*resource.Quantity` rather than `resource.Quantity` (in the 3 cluster.go files) so to keep the same behaviour (spec) for existing clusters, unless explicitly set by the user. It will also pick out updates required to some tests as the nil values should be dropped.

Reply:
In general, in my projects, I normally avoid the use of pointers because those nils always seem to come back to bite you, or require extra handling at some point.  This is not my project, though :-)

I tried to match the existing behavior by editing code in other places.  Check `protokube/pkg/protokube/etcd_cluster.go` line 194... `if c.MemoryRequest.IsZero() {` is used to determine if the user did not specify a quantity.  The existing defaults are used in that case.

Are you sure that the user's clusters will change if nothing was specified?  I specifically tried to write the code in such a way that users could ignore the option and functionality would not change.

Thank you for taking the time to look at this big PR!  It's a simple change, but many parts of the code are effected.

Reply:
Understood!  It is a shame that the `resource.Quantity` does not marshal in such a way that `omitempty` will take effect.  I will work on switching these into pointers.

Reply:
@gambol99 this is done.